### PR TITLE
Fix PackageManifest Server Crashes

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description

Didn't realize that `GetLabels()` returns a reference, not a copy.

Addresses https://jira.coreos.com/browse/OLM-1111

### CODE REVIEW

File: pkg/package-server/provider/registry.go [hunk 1]
```diff
@@ -289,9 +289,11 @@ func toPackageManifest(pkg *api.Package, client registryClient) (*operators.Pack
 	catsrc := client.source
 	manifest := &operators.PackageManifest{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:              pkg.GetName(),
-			Namespace:         catsrc.GetNamespace(),
-			Labels:            catsrc.GetLabels(),
+			Name:      pkg.GetName(),
+			Namespace: catsrc.GetNamespace(),
+			Labels: labels.CloneAndAddLabel(
```
Comment (ecordell) [inline pkg/package-server/provider/registry.go]: Why clone twice?

Comment (alecmerdler) [reply pkg/package-server/provider/registry.go]: So it can all be done in-place without extra variables. There isn't a `CloneAndAddLabels()` function.

**Pull request merged**